### ISSUE TITLE
renaming a chroot: sudo edit-chroot -m old_name new_name -- fails

### ISSUE THREAD

**Issue by Watney** [dnschneid/crouton#1463]
I'm using ``croutonversion``
crouton: version 1-20150301132939~master:80da699e
release: trusty
architecture: amd64
xmethod: xorg
targets: xorg,keyboard,extension,xiwi,xfce
host: version 6812.3.0 (Official Build) dev-channel glimmer_freon  

When I try to rename chroot ``test`` to ``trusty1`` it report
``chronos@localhost / $ sudo edit-chroot -m test trusty1``
``/mnt/stateful_partition/crouton/chroots/trusty1 not found.``

**Comment by dnschneid** [comment]
The syntax is `sudo edit-chroot -m dest src`.  You can also do `sudo edit-chroot src -m dest` if that is a bit clearer.

**Comment by Watney** [comment]
Got it. Thanks so much @dnschneid!!

**Comment by DennisLfromGA** [comment]
> The syntax is sudo edit-chroot -m dest src. You can also do sudo edit-chroot src -m dest if that is a bit clearer.

Didn't know that was possible, thanx for the tip. It seems more logical and conventional to me.

**Comment by Watney** [comment]
@DennisLfromGA What would you think about a cheatsheet in the Wiki that lists the common commands?

**Comment by DennisLfromGA** [comment]
@Watney - Sounds like it might be useful - go for it!

**Comment by Watney** [comment]
@DennisLfromGA what this needs is some adult supervision! I've got a list, but I'm a n00b and it's all second hand, stuff I've copied from others. 

On the minus side, a list wouldn't promote user learning and inquiry (e.g., [the Arch Way](https://wiki.archlinux.org/index.php/The_Arch_Way)) if that's important in this community. 

I'll defer to others. I'm glad to start an entry, but someone who knows what s/he's doing, needs to clean up after me, in other words, an adult.

**Comment by divx118** [comment]
@Watney Go for it. There will always be someone to look it over. @DennisLfromGA , someone else, or me.

**Comment by Watney** [comment]
@divx118 @DennisLfromGA It's live [here](https://github.com/dnschneid/crouton/wiki/Crouton-Command-Cheat-Sheet)

**Comment by DennisLfromGA** [comment]
@Watney,

> I've got a list, but I'm a n00b and it's all second hand, stuff I've copied from others.

Yup, I wasn't sure exactly what you had in mind when you mentioned some common commands, that's why I deferred it to you. If you've got a list then that sounds like a good place to start.

Like @divx118 said, we'll be glad to look over and pitch in on this project so don't worry about it.
Just getting started is the hardest part. ;)

**Comment by DennisLfromGA** [comment]
That looks *really* good, I don't think you need much 'adult' supervision at all.
We'll pore over it and see if we can add anything that might be of value.

Again, good job! :+1:

**Comment by Watney** [comment]
@DennisLfromGA Thanks!! 

Now, here's where I struggle: switching between xiwi and xorg, using the commands that crouton understands. The documentation mentions using the switch -X, but I've never been able to get it to work. I copied my method from you on G+ as I recall! :)

**Comment by divx118** [comment]
@Watney -X should work. For example `sudo startxfce4 -X xiwi` Also for running crouton in a window you only need to specify `-t xiwi`, no need to add extension to it. Target xiwi automatically pulls in extension.

**Comment by tedm** [comment]
@divx118 Maurice, a few days ago, installing a new  precise / xfce / xiwi chroot on the ARM didn't pull in extension, but did on trusty. I know precise may go away soon, but I'll try it again in a few days.

**Comment by tedm** [comment]
@Watney on the ARM, you can't really switch once xiwi is installed, unless there's something I've missed recently, you have to install your chroot again to get xephyr. You are correct that there is no xorg on ARM.

**Comment by divx118** [comment]
@tedm I doubt it, just try it again. If it really isn't pulling in the extension then it is a bug and there should be an issue created for it, However like I said I doubt it will fail pulling in the extension.

**Comment by Watney** [comment]
@divx118 Thanks, however, this doesn't work: ``sudo enter-chroot -n chrootname -X xiwi`` but this does work: ``sudo enter-chroot -n chrootname env XMETHOD=xiwi startxfce4``

Maybe it's a bug?

**Comment by dnschneid** [comment]
Try `sudo startxfce4 -X xiwi`.

**Comment by Watney** [comment]
@dnschneid yes, ``sudo startxfce4 -X xiwi`` works, but if I want to use a chroot other than the default (i.e., first one in the list) with the ``-n`` switch, it fails

**Comment by divx118** [comment]
@Watney You can use -n after a startscript. So `sudo startxfce4 -X xiwi -n trusty`. Also the long way `sudo enter-chroot -n trusty -X xiwi startxfce4` also works.

**Comment by Watney** [comment]
@divx118 presto! That works. Now you know why I wanted a cheat sheet in the wiki. Thanks so much

**Comment by divx118** [comment]
@Watney What I forgot to mention is that -X only sets the environmental variable XMETHOD, so it doesn't start anything, that is why `sudo enter-chroot -n trusty -X xiwi` "failed" You didn't start any DE. You need to add startunity or startxfce4 after it. Better however is to promote the standard startscripts like `sudo startxfce4 -n trusty -X xiwi`

**Comment by tedm** [comment]
@dnschneid Can you ELI5 again why we can't -x xephyr , was it because it's going soon? @divx118 will know in a few minutes, installing a new precise chroot with -t xiwi,xfce if it brings in extension, I'll then try xfce,xiwi just to see if ordering still maters. Thanks.

**Comment by dnschneid** [comment]
Well, if you have xephyr (successfully) installed, you're free to use it.  At this point, though, the only platforms where it makes sense to install xephyr are Samsung ARM devices on any channel and Tegra K1 devices on M41 and below (that's beta and stable channels at the moment).  Those platforms will automatically install xephyr if you don't explicitly specify xiwi.

As Chromium OS continues to move away from X11 (via the Freon display manager), xephyr will work on fewer and fewer systems, so it's better to only use it if xorg doesn't work.

**Comment by tedm** [comment]
@dnschneid OK, thanks, and I just read #1265 which was closed a few days ago. So basically moving forward, xiwi for crouton is the way to go for the Samsung ARMs, and I'm fine with this. Someday when I make big money, I want to get a newer chromebook, but really prefer fanless designs. What's out there these days? @divx118 Maurice, precise -t xiwi,xfce doesn't bring in extension. I filed an issue for that.

**Comment by DennisLfromGA** [comment]
Just as an experiment, on my HP Chromebook 14, I've installed xorg, xiwi, and xephyr.
I can activate any of the three xmethods using the -X option to my start* scripts and they all open and run flawlessly.

**Comment by tedm** [comment]
@DennisLfromGA Thanks, I can as well after installing the xephyr target on the xiwi,xfce chroot, but after adding xephyr, it gets put in /etc/crouton/xmethod as the default, so xiwi will always need a "start* -X xiwi" to startup windowed, but this works pretty well.

Crouton automatically used xephyr by default (no xorg) before xiwi on the ARMs, so I got used to it just being there. Thanks!

**Comment by DennisLfromGA** [comment]
I think you could rearrange the xmethods in `/etc/crouton/targets` to put xiwi first in line.
You could also manually edit `/etc/crouton/xmethod` to replace 'xephyr` with 'xiwi'.
But I'm sure neither of those procedures is recommended; I suppose the recommende way of reordering them is to just do an update specifying `-t xiwi` again.
